controllers/prep_handlers.go

```diff
@@ -442,37 +413,86 @@ func (r *ImageBasedUpgradeReconciler) handlePrep(ctx context.Context, ibu *lcav1
 		return
 	}
 
+	r.Log.Info("Current PrepTask status",
+		"IsActive", r.PrepTask.Active,
+		"IsSuccess", r.PrepTask.Success,
+		"PrecacheRequested", r.Precache.PrecacheRequested)
+
 	switch {
 	case !r.PrepTask.Active:
 		r.PrepTask.done = make(chan struct{})
 		r.PrepTask.Active = true
 		r.PrepTask.Success = false
 		r.PrepTask.Progress = "Prep stage initialized"
+		r.PrepTask.AdditionalComplete = ""
+		r.Precache.PrecacheRequested = false
 		go func() {
 			err = r.prepStageWorker(ctx, ibu)
-			close(r.PrepTask.done)
 			if err != nil {
 				r.Log.Error(err, "Prep stage failed with error")
 				r.PrepTask.Success = false
+				close(r.PrepTask.done)
+				return
+			}
+			if r.Precache.PrecacheRequested {
+				// this indicates we can now safely check for precache job status during reconcile
+				r.PrepTask.Progress = "Prep in progress, watching precache job"
+				r.Log.Info("Precache requested")
 			} else {
-				r.Log.Info("Prep stage completed successfully!")
+				r.Log.Info("No precache requested")
```

Comment:
If `prepStageWorker` func is executed successfully, a precache job must have been created. If not, the prep will just fail. It seems we don't have a "No precache requested" case, should we remove it? It looks like r.Preache.PreacheRequested adds the complexity.

Reply:
Dropped `PreacheRequested`